internal/libyaml/api_test.go

```diff
@@ -0,0 +1,527 @@
+	if len(methodList) == 0 {
+		panic("empty method list")
+	}
+
+	methodName, ok := methodList[0].(string)
+	if !ok {
+		panic(fmt.Sprintf("method name should be string, got %T", methodList[0]))
+	}
+	args := methodList[1:]
+
+	v := reflect.ValueOf(obj)
+	if v.Kind() == reflect.Ptr {
+		v = v.Elem()
+	}
+
+	m := v.Addr().MethodByName(methodName)
+	if !m.IsValid() {
+		panic("method not found: " + methodName)
+	}
+
+	methodType := m.Type()
+
+	// Build argument list
+	var callArgs []reflect.Value
+	for i, arg := range args {
+		paramType := methodType.In(i)
+
+		// Handle different parameter types
+		if paramType.Kind() == reflect.Bool {
+			val := parseBoolArg(arg)
+			callArgs = append(callArgs, reflect.ValueOf(val))
+		} else if paramType.Kind() == reflect.Slice && paramType.Elem().Kind() == reflect.Uint8 {
+			// Byte slice parameter
+			str := parseStringArg(arg, useBytes)
+			callArgs = append(callArgs, reflect.ValueOf([]byte(str)))
+		} else {
+			// Try parsing as constant/int
+			val := parseArg(arg)
+			convertedVal := reflect.ValueOf(val).Convert(paramType)
+			callArgs = append(callArgs, convertedVal)
+		}
+	}
+
+	// Call the method
+	m.Call(callArgs)
+}
+
+// parseArg parses an argument which could be int, bool, or string constant
+func parseArg(arg interface{}) int {
+	switch v := arg.(type) {
+	case int:
+		return v
+	case string:
+		if looksLikeConstant(v) {
+			return parseConstant(v)
+		}
+		// Try parsing as int
+		if val, err := strconv.Atoi(v); err == nil {
+			return val
+		}
+		panic(fmt.Sprintf("cannot parse arg as int: %v", arg))
+	default:
+		panic(fmt.Sprintf("unsupported arg type: %T", arg))
+	}
+}
+
+// parseBoolArg parses a boolean argument
+func parseBoolArg(arg interface{}) bool {
+	switch v := arg.(type) {
+	case bool:
+		return v
+	case string:
+		return v == "true"
+	default:
+		panic(fmt.Sprintf("cannot parse arg as bool: %v", arg))
+	}
+}
+
+// parseStringArg parses a string argument
+func parseStringArg(arg interface{}, useBytes bool) string {
+	if str, ok := arg.(string); ok {
+		return str
+	}
+	panic(fmt.Sprintf("cannot parse arg as string: %v", arg))
+}
+
+// looksLikeConstant checks if a string looks like a constant name
+func looksLikeConstant(s string) bool {
+	if len(s) == 0 {
+		return false
+	}
+	// Check if it's all uppercase letters, digits, and underscores
+	for _, c := range s {
+		if !((c >= 'A' && c <= 'Z') || (c >= '0' && c <= '9') || c == '_') {
+			return false
+		}
+	}
+	return true
+}
+
+// parseConstant parses a constant name to its integer value
+func parseConstant(name string) int {
+	// Handle boolean values
+	if name == "true" {
+		return 1
+	}
+	if name == "false" {
+		return 0
+	}
+
+	// Try parsing as int first
+	if val, err := strconv.Atoi(name); err == nil {
+		return val
+	}
+
+	// Use IntOrStr to parse other constants
+	ios := IntOrStr{}
+	err := ios.FromValue(name)
+	if err != nil {
+		panic(fmt.Sprintf("failed to parse constant %q: %v", name, err))
```

Comment:
As mentioned before, here and everywhere, mrgjid should receive a *testing.T, call t.Helper() and use t.Fatalf in such conditions

Panic should be avoided